search/searcher/search_phrase.go

```diff
@@ -102,39 +102,42 @@ func (s *PhraseSearcher) Next(ctx *search.SearchContext) (*search.DocumentMatch,
 		freq := 0
 		firstTerm := s.terms[0]
 		for field, termLocMap := range s.currMust.Locations {
-			rvtlm := make(search.TermLocationMap, 0)
 			locations, ok := termLocMap[firstTerm]
-			if ok {
-			OUTER:
-				for _, location := range locations {
-					crvtlm := make(search.TermLocationMap, 0)
-				INNER:
-					for i := 0; i < len(s.terms); i++ {
-						nextTerm := s.terms[i]
-						if nextTerm != "" {
-							// look through all these term locations
-							// to try and find the correct offsets
-							nextLocations, ok := termLocMap[nextTerm]
-							if ok {
-								for _, nextLocation := range nextLocations {
-									if nextLocation.Pos == location.Pos+float64(i) && nextLocation.SameArrayElement(location) {
-										// found a location match for this term
-										crvtlm.AddLocation(nextTerm, nextLocation)
-										continue INNER
-									}
-								}
-								// if we got here we didn't find a location match for this term
-								continue OUTER
-							} else {
-								continue OUTER
-							}
+			if !ok {
+				continue
+			}
+
+			rvtlm := make(search.TermLocationMap, 0)
+
+		OUTER:
+			for _, location := range locations {
+				crvtlm := make(search.TermLocationMap, 0)
+			INNER:
+				for i := 1; i < len(s.terms); i++ {
+					nextTerm := s.terms[i]
+					if nextTerm == "" {
+						continue
+					}
+					// look through all these term locations
+					// to try and find the correct offsets
+					nextLocations, ok := termLocMap[nextTerm]
+					if !ok {
+						continue OUTER
+					}
+					for _, nextLocation := range nextLocations {
+						if nextLocation.Pos == location.Pos+float64(i) && nextLocation.SameArrayElement(location) {
+							// found a location match for this term
+							crvtlm.AddLocation(nextTerm, nextLocation)
```

Comment:
I'm not as familiar with the phrase searcher, but to my eyes, this change looks almost functionally the same, except with the change/optimization where the INNER for-loop starts at 1 instead of 0.

On that, does that mean the crvtlm.AddLocation(nextTerm, nextLocation) here won't happen for the first (or 0'th) term?  And, if so, is that an important difference in behavior?